driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -253,7 +253,7 @@ private boolean requireOpMsgResponse() {
         if (responseExpected) {
```

Comment:
We can simplify this class bit further by adding the following assertion in the constructor:

```
assertFalse(getOpCode() == OpCode.OP_QUERY && !responseExpected);
```

this assertion should never fire because OP_QUERY effectively is only used for the initial `hello` command on a new connection, which should always be expecting a response.

Then we can remove the `!useOpMsg()` condition in `isResponseExpected`, and inline this method.